packages/jest-config/src/__tests__/normalize.test.js

```diff
@@ -303,6 +303,24 @@ describe('transform', () => {
       ['abs-path', '/qux/quux'],
     ]);
   });
+  it('pulls in config if its passed as an array', () => {
+    const {options} = normalize(
+      {
+        rootDir: '/root/',
+        transform: {
+          [DEFAULT_CSS_PATTERN]: '<rootDir>/node_modules/jest-regex-util',
+          [DEFAULT_JS_PATTERN]: ['babel-jest', {rootMode: 'upward'}],
+          'abs-path': '/qux/quux',
+        },
+      },
+      {},
+    );
+    expect(options.transform).toEqual([
+      [DEFAULT_CSS_PATTERN, '/root/node_modules/jest-regex-util'],
+      [DEFAULT_JS_PATTERN, require.resolve('babel-jest'), {rootMode: 'upward'}],
```

Comment:
I wonder if we pay any perf penalty for making this an array of variable length (previously always a tuple, now an array of 2 or 3 items). It's accessed before every file transform, so definitely something worth double-checking.

Reply:
wouldn't it be uniform in most cases that are using only one config file?